tools/importer-msgraph-metadata/internal/pipeline/task_translate_service.go

```diff
@@ -0,0 +1,179 @@
+		if _, ok := sdkServices[resource.Service]; !ok {
+			sdkServices[resource.Service] = sdkModels.Service{
+				APIVersions:         make(map[string]sdkModels.APIVersion),
+				Generate:            true,
+				ResourceProvider:    nil,
+				TerraformDefinition: nil,
+			}
+		}
+
+		if _, ok := sdkServices[resource.Service].APIVersions[resource.Version]; !ok {
+			sdkServices[resource.Service].APIVersions[resource.Version] = sdkModels.APIVersion{
+				Generate:  true,
+				Preview:   normalize.VersionIsPreview(resource.Version),
+				Resources: make(map[string]sdkModels.APIResource),
+				Source:    sdkModels.MicrosoftGraphMetaDataSourceDataOrigin,
+			}
+		}
+
+		if _, ok := sdkServices[resource.Service].APIVersions[resource.Version].Resources[resource.Category]; !ok {
+			sdkServices[resource.Service].APIVersions[resource.Version].Resources[resource.Category] = sdkModels.APIResource{
+				Constants:   make(map[string]sdkModels.SDKConstant),
+				Models:      make(map[string]sdkModels.SDKModel),
+				Operations:  make(map[string]sdkModels.SDKOperation),
+				ResourceIDs: make(map[string]sdkModels.ResourceID),
+			}
+		}
+
+		serviceModels := make(parser.Models)
+
+		// Populate everything else
+		for _, operation := range resource.Operations {
+			var resourceIdName *string
+
+			if operation.ResourceId != nil {
+				resourceIdName = &operation.ResourceId.Name
+
+				sdkResourceId, err := operation.ResourceId.DataApiSdkResourceId()
+				if err != nil {
+					return nil, err
+				}
+
+				sdkServices[resource.Service].APIVersions[resource.Version].Resources[resource.Category].ResourceIDs[operation.ResourceId.Name] = *sdkResourceId
+			}
+
+			var requestObject *sdkModels.SDKObjectDefinition
+			requestObjectIsCommonType := true
+
+			if operation.RequestModel != nil {
+				if !models.Found(*operation.RequestModel) {
+					return nil, fmt.Errorf("request model not found")
+				}
+
+				if model := models[*operation.RequestModel]; !model.IsValid() {
+					return nil, fmt.Errorf("request model was invalid")
+				} else if !model.Common {
+					requestObjectIsCommonType = false
+
+					if err := serviceModels.MergeDependants(models, *operation.RequestModel); err != nil {
+						return nil, err
+					}
+				}
+
+				requestObject = &sdkModels.SDKObjectDefinition{
+					ReferenceName:             operation.RequestModel,
+					ReferenceNameIsCommonType: &requestObjectIsCommonType,
+					Type:                      sdkModels.ReferenceSDKObjectDefinitionType,
+				}
+			} else if operation.RequestType != nil {
+				requestObject = &sdkModels.SDKObjectDefinition{
+					Type: operation.RequestType.DataApiSdkObjectDefinitionType(),
+				}
+			}
+
+			var responseObject *sdkModels.SDKObjectDefinition
+			responseObjectIsCommonType := true
+
+			for _, response := range operation.Responses {
+				if response.Type != nil && *response.Type == parser.DataTypeModel && response.ModelName != nil {
+					if !models.Found(*response.ModelName) {
```

Comment:
in errors like this we should highlight _which_ model can't be found? would mean you have the information needed to investigate this without needing to trigger an import manually, when something goes wrong?